### DISCUSSION TITLE
Tons of new suggestions

### DISCUSSION THREAD

**Discussion by Cool-Scratcher** [ScratchAddons/ScratchAddons#754] [Ideas]
### Where these suggestions came from
I found this excellent Scratch project called "[SCRATCH UI 4.0](https://scratch.mit.edu/projects/443561067/)" that introduces a whole new concept of how Scratch's UI might change. The thing that caught my attention, though, is the functionality introduced. As well as being excellent, the ideas made sense and looked like they might work.

Some ideas, such as the "show possible bugs" in the editor would be extremely hard to implement. The ones that are possible, however, I put here.

Most of these ideas were thought of by [design-ileb7](https://scratch.mit.edu/users/design-ileb7/) on Scratch.

### The actual suggestions

- Integrating Scratch Addons and [ScratchStats](https://scratchstats.com/). For example, on the home page, there could be a link to the projects people like the most.
- A built-in Scratch PFP (SPFP) that generates a profile picture. Basically, choosing a background, choosing the main image, and then choosing overlays.
- From the main website, you can set a follower goal and see on average, how many people follow you each day. You can see your follower goal visualized on a bar, and the newest comment made.
- What's Happening? has a button to show more.
- Hovering over a project brings up a quick stats window to see the title, author, date shared, number of blocks, love/view ratio, and likes/faves ratio.
- Explore page has filters, like by country, type of project, and studios.
- Also a "Similar to..." section.
- The "My Stuff" page has so many changes that it would be impossible to put them in words. Therefore, here you go:
![mystuffpage-new](https://user-images.githubusercontent.com/65724251/99189024-0b358300-272d-11eb-898e-ca29413c96f1.png)
(All credit goes to [design-ileb7](https://scratch.mit.edu/users/design-ileb7/) on Scratch.)
- I'd like to draw attention to having multiple accounts managed here. Adding alts? That sounds great. However, it would be very difficult, seeing as it is more backend than frontend. It's just an idea.
- What about being able to mute push notifications for a certain amount of time?
- Quick accept/decline of studio invitations from people!
- From the editor, you can send collab invitations to other Scratchers.
- You can also send projects to your alt account.
- The paint editor allows you to blur, round corners, and convert to .png.
- As [design-ileb7](https://scratch.mit.edu/users/design-ileb7/) says:

> The new profile allows you to customize your profile however
> you want. For example with a custom background and 
> different categories that are showing your new project
> or a text about you or for example your shoutouts.
> You can add a maximum of 10 categories to your profile.

- From the project page, you can follow the author of the project, view statistics, and view scripts. In the description, you can make titles, headers, text, tags, or credits. You can change the color, boldness, or size, underline or strikethrough, change the font or add separate sections with Markdown.
- Comments have Markdown styling too.

Thanks for reading this.

**Comment by TheColaber** [comment]
- K, "What's Happening? has a button to show more." Has been done in the "Show full areas" addon.
- For "Quick accept/decline of studio invitations from people!" This can be done with this API: `https://scratch.mit.edu/site-api/users/curators-in/STUDIO_ID/add/?usernames=USERNAME`
- For "From the editor, you can send collab invitations to other Scratchers." Ok, that is definitely impossible... It is impossible to have collaboration unless it's a mod.
- For "You can also send projects to your alt account." I don't really think that would be possible...
- For "The paint editor allows you to blur, round corners, and convert to .png." To convert to png, convert to bitmap... bitmap is png. Now about blurring and rounding corners, I am not really sure how to access the canvas, but if anyone knows, then that is possible.
- For "You can change the color, boldness, or size, underline or strikethrough, change the font or add separate sections with Markdown." and "Comments have Markdown styling too." this is possible, but this would only work for people who both have ScratchAddons AND have the addon enabled. For other users, it would look weird like this: `[b]This is bold[/b]`

**Comment by Explosion-Scratch** [comment]
Collab invitations would be hard, same with blur in the editor.I don't know about adding alt accounts, it may be hard. Also a similar to section would be nice but likely hard to implement.We would need server room for SA to make profiles customize-able and people without the addon would not be able to see it. The others though sound great! I would also love a love to view ratio, and also a graph of follows on a user's profile would be cool!

**Comment by retronbv** [comment]
most of this stuff is being made by [SC4](https://github.com/Scratch-Client-4)
